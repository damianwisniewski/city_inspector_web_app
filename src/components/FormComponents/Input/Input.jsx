import React from 'react'
import PropTypes from 'prop-types'

const Input = ({ id, placeholder, children, floatingLabel, ...rest }) => (
	<div className='input-wrapper'>
		<input
			{...rest}
			data-animate={floatingLabel}
			className='input-wrapper__field'
			id={id}
			placeholder={placeholder}
		/>
		<label className='input-wrapper__label' htmlFor={id}>
			{children}
		</label>
	</div>
)

Input.propTypes = {
	id: PropTypes.string.isRequired,
	placeholder: PropTypes.string.isRequired,
	children: PropTypes.string.isRequired,
	floatingLabel: PropTypes.bool.isRequired,
}

export default Input
