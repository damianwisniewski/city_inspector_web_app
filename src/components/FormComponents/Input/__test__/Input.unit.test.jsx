import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Input from '../Input'

configure({ adapter: new Adapter() })

describe('<Input />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let InputElement
	const InputText = 'mockText'
	const id = 'mockId'
	const placeholder = 'mock placeholder'

	test('should render correctly Input, with floatingLabel value equal true', () => {
		InputElement = shallow(
			<Input floatingLabel placeholder={placeholder} id={id} label={InputText} />,
		)
		expect(toJson(InputElement)).toMatchSnapshot()
	})

	test('should render correctly Input, with floatingLabel value equal false', () => {
		InputElement = shallow(
			<Input labelType='basic' placeholder={placeholder} id={id} label={InputText} />,
		)
		expect(toJson(InputElement)).toMatchSnapshot()
	})
})
