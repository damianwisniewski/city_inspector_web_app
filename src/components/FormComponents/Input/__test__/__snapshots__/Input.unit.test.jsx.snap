// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly Input, with floatingLabel value equal false 1`] = `
<div
  className="input-wrapper input-wrapper--basic"
>
  <input
    className="input-wrapper__field"
    data-animate={false}
    id="mockId"
    placeholder="mock placeholder"
  />
  <label
    className="input-wrapper__label input-wrapper__label--basic"
    htmlFor="mockId"
  >
    mockText
  </label>
</div>
`;

exports[`<Input /> should render correctly Input, with floatingLabel value equal true 1`] = `
<div
  className="input-wrapper input-wrapper--basic"
>
  <input
    className="input-wrapper__field"
    data-animate={false}
    floatingLabel={true}
    id="mockId"
    placeholder="mock placeholder"
  />
  <label
    className="input-wrapper__label input-wrapper__label--basic"
    htmlFor="mockId"
  >
    mockText
  </label>
</div>
`;
